navigator/frontend/src/ui-core/src/api/DamlLfValue.ts

```diff
@@ -124,9 +138,25 @@ export function evalPath(value: DamlLfValue, path: string[], index: number = 0):
         return value;
       } else if (path[index] === value.constructor) {
         return evalPath(value.value, path, index + 1);
-      } else {
+      } else {
         return notFound;
       }
+    case 'map':
+      if (isLast) {
+        return value;
+      } else {
+        const key = path[index];
+        if (value.value === null) {
```

Comment:
Can the value ever be `null`? `null` is only used in the serialization of `Optional`, and `value` is typed as `DamlLfValueMapEntry[]`, which is unrelated to `null`. I'm actually surprised the compiler does not complain here.